src/pymor/algorithms/to_matrix.py

```diff
@@ -133,13 +137,7 @@ def action_ComponentProjectionOperator(self, op):
 
     @match_class(ConcatenationOperator)
     def action_ConcatenationOperator(self, op):
-        mats = [self.apply(o) for o in op.operators]
-        while len(mats) > 1:
-            if self.format is None and not sps.issparse(mats[-2]) and sps.issparse(mats[-1]):
-                mats = mats[:-2] + [mats[-1].T.dot(mats[-2].T).T]
-            else:
-                mats = mats[:-2] + [mats[-2].dot(mats[-1])]
-        return mats[0]
+        return op.as_range_array().to_numpy().T
```

Comment:
```suggestion
        return op.as_range_array(mu=self.mu).to_numpy().T
```

Also, I see now that it might be good to keep the old implementation in case the result is a sparse matrix. Something like the following should work:

```python
try:
    mats = [self.apply(o) for o in op.operators]
    from functools import reduce
    from operator import matmul
    return reduce(matmul, mats)
except NoMatchingRuleError:
    return op.as_range_array(mu=self.mu).to_numpy().T
```